client_it_test.go

```diff
@@ -213,3 +214,95 @@ func TestClient_AddDistributedObjectListener(t *testing.T) {
 		mu.Unlock()
 	})
 }
+
+func TestClusterReconnection(t *testing.T) {
+	ctx := context.Background()
+	cls := it.StartNewClusterWithOptions("go-cli-test-cluster", 15701, 3)
+	mu := &sync.Mutex{}
+	events := []hz.LifecycleState{}
+	config := cls.DefaultConfig()
+	config.AddLifecycleListener(func(event hz.LifecycleStateChanged) {
+		mu.Lock()
+		events = append(events, event.State)
+		mu.Unlock()
+	})
+	c, err := hz.StartNewClientWithConfig(ctx, config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(5 * time.Second)
+	cls.Shutdown()
+	time.Sleep(5 * time.Second)
+	cls = it.StartNewClusterWithOptions("go-cli-test-cluster", 15701, 3)
+	time.Sleep(5 * time.Second)
+	cls.Shutdown()
+	c.Shutdown(ctx)
+	mu.Lock()
+	defer mu.Unlock()
```

Comment:
Do we need these two lines? Stopping the user event dispatcher prevents events' processing, does not it? I deleted these lines and test still passed?

Reply:
Try with `go test -run TestClusterReconnection -race .` ;)